aioftp/server.py

```diff
@@ -1567,8 +1569,96 @@ def get_paths(self, connection, path):
         return passive_server
 
     @ConnectionConditions(ConnectionConditions.login_required)
+    async def port(self, connection, rest):
+
+        if connection.is_passive:
+
+            if connection.future.data_connection.done():
+
+                connection.data_connection.close()
+
+            else:
+
+                connection.future.data_connection.cancel()
+
+            del connection.data_connection
+
+            if connection.future.passive_server.done():
+
+                connection.passive_server.close()
+
+            else:
+
+                connection.future.passive_server.cancel()
+
+            del connection.passive_server
+
+            connection.is_passive = False
+
+        async def open_data_connection(host, port):
+
+            reader, writer = await asyncio.open_connection(
+                host=host,
+                port=port,
+                loop=connection.loop
+            )
+
+            return ThrottleStreamIO(
+                reader,
+                writer,
+                throttles=connection.command_connection.throttles,
+                timeout=connection.socket_timeout,
+                loop=connection.loop
+            )
+
+        nums = tuple(map(int, str.split(rest, ",")))
+        host = str.join(".", map(str, nums[:4]))
+        port = (nums[4] << 8) | nums[5]
+
+        if not connection.future.passive_server.done():
```

Comment:
we need `connection.futures.passive_server` to pass `ConnectionCondition(passive_server_started)` checks